src/back/Calculator.cc

```diff
@@ -0,0 +1,274 @@
+    else if (token == "(")
+      stack_token_.push(queue_.front());
+    else if (token == ")") {
+      if (stack_token_.top().GetName() == "(") {
+        stack_token_.pop();
+        queue_.pop();
+        continue;
+      } else
+        while (stack_token_.top().GetName() != "(")
+          Counting();
+      stack_token_.pop();
+    } else {
+      if (stack_token_.empty())
+        stack_token_.push(queue_.front());
+      else
+        Conditions();
+    }
+    if (!queue_.empty())
+      queue_.pop();
+  }
+  while (!stack_token_.empty())
+    Counting();
+}
+
+void s21::Calculator::Counting() {
+  std::visit(overloaded{[&](unary_function function) {
+                          PushNumberToStack(function(PopFromResult()));
+                        },
+
+                        [&](binary_function function) {
+                          double right_argument = PopFromResult();
+                          double left_argument = PopFromResult();
+                          PushNumberToStack(
+                              function(left_argument, right_argument));
+                        },
+
+                        [&](auto) {}},
+
+             stack_token_.top().GetFunction());
+  stack_token_.pop();
+}
+
+void s21::Calculator::Conditions() {
+  int lastToken = stack_token_.top().GetPriority();
+  int currentToken = queue_.front().GetPriority();
+
+  if (lastToken > currentToken || lastToken == currentToken) {
+    Counting();
+    stack_token_.push(queue_.front());
+  } else
+    stack_token_.push(queue_.front());
+}
+
+void s21::Calculator::SetAnswer() {
+  answer_ = stod(stack_number_.top().GetName());
+  stack_number_.pop();
+}
+
+double s21::Calculator::PopFromResult() {
+  if (stack_number_.empty())
+    throw std::string("NOT ENOUGH NUMBERS");
+
+  s21::Token number = stack_number_.top();
+  std::string name = number.GetName();
+  stack_number_.pop();
+  return stod(name);
+}
+
+void s21::Calculator::PushNumberToStack(double value) {
+  std::ostringstream token;
+  token << value;
+  std::string name = token.str();
+  Token result(name, kDefault, kNone, kNumber, value);
+  stack_number_.push(result);
+}
+
+void s21::Calculator::PushTokenToQueue(std::string input) {
+  auto found_token = token_map_.find(input);
+  if (found_token == token_map_.end())
+    throw std::string("INVALID CHARACTER(S)");
+  queue_.push(found_token->second);
+}
+
+void s21::Calculator::ClearСoordinates() {
+  coordinates_x_.clear();
+  coordinates_y_.clear();
+};
+
+void s21::Calculator::CalculateGraph(std::string input, std::string input_x) {
+  ClearСoordinates();
+  double dot = stod(input_x);
+  for (double x = -dot; x <= fabs(dot); x += 0.01) {
+    CalculateAnswer(input, std::to_string(x));
+    coordinates_x_.push_back(x);
+    coordinates_y_.push_back(GetAnswer());
+  }
+}
+
+std::vector<double> s21::Calculator::GetCoordinatesX() {
+  return coordinates_x_;
+}
+std::vector<double> s21::Calculator::GetCoordinatesY() {
+  return coordinates_y_;
+}
```

Comment:
Лишнее копирование - возвращай константную ссылку, если оставишь этот вариант.
```suggestion
const std::vector<double>& s21::Calculator::GetCoordinatesX() {
  return coordinates_x_;
}
const std::vector<double>& s21::Calculator::GetCoordinatesY() {
  return coordinates_y_;
}
```